### DISCUSSION TITLE
Can't Connect Postgres to Budibase Using Docker

### DISCUSSION THREAD

**Discussion by jonasp23** [Budibase/budibase#11015] [Help] [answered]
Hi
I've been trying for days to connect my docker instance of Postgres but nothing is working and I always get the same error. `"Unable to connect - getaddrinfo ENOTFOUND host.docker.internal"`

Yes, I am using host.docker.internal as my host, user and password are correct. 
I have managed to connect the docker instance to pgAdmin so I know the connection is working. Is there anything else I am missing, I've tried so many things, even changing ports , different host names etc..

Please let me know if you need any information from me.

TIA

**Comment by ConorWebb96** [comment, marked as answer]
Hi @jonasp23,

When you say docker instance do you mean a container you've created in docker for your Postgres db? If so I've supplied a screenshot of my settings used to connect to docker db within Buidbase. 
<img width="571" alt="Screenshot 2023-06-26 at 07 37 30" src="https://github.com/Budibase/budibase/assets/126772285/5f345015-f61b-422d-94df-68951290e7d7">

There are a few reasons that may be affecting your ability to connect. I believe its more than likely due to how you setup this docker container. Below is the command I've run to setup my postgres instance.
`docker run --name some-postgres -p 5432:5432 -e POSTGRES_PASSWORD=passwordgoeshere -d postgres`
<img width="260" alt="Screenshot 2023-06-26 at 07 57 36" src="https://github.com/Budibase/budibase/assets/126772285/c4dacd63-5e88-4f91-ba1e-62d764792263">

Make sure the Database you are trying to connect to exists and that the overall postgres instance is running on docker..
<img width="986" alt="Screenshot 2023-06-26 at 07 38 51" src="https://github.com/Budibase/budibase/assets/126772285/102c2923-180e-46c7-b60c-e37007c7a743">

I suggest making sure it has a port assigned to it, I noticed when setting up my instance some of the guides didn't provide a port to the creation command of this Postgres container. I think it defaults to 5432, however this only happens on initial setup after that you have to supply the port you want to use to the command otherwise it will be created without one.

I hope this helps.

**Reply by MrChadMWood** [reply]
Why cant Budibase refer to the docker service name as the host? That is the standard when apps share a docker network.

**Reply by MrChadMWood** [reply]
For any future comers using compose, the `-e` is exposing the port here.
Inside compose, this would look like:
```
    expose:
      - "5432"
```
